zhaquirks/tuya/ts110e.py

```diff
@@ -0,0 +1,178 @@
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+from zhaquirks.tuya import (
+    NoManufacturerCluster,
+    TuyaDimmerSwitch,
+    TuyaZBExternalSwitchTypeCluster,
+)
+
+TUYA_LEVEL_ATTRIBUTE = 0xF000
+TUYA_BULB_TYPE_ATTRIBUTE = 0xFC02
+TUYA_MIN_LEVEL_ATTRIBUTE = 0xFC03
+TUYA_MAX_LEVEL_ATTRIBUTE = 0xFC04
+TUYA_CUSTOM_LEVEL_COMMAND = 0x00F0
+
+
+class TuyaLevelPayload(t.Struct):
+    """Tuya Level payload."""
+
+    level: t.uint16_t
+    transtime: t.uint16_t
+
+
+class TuyaBulbType(t.enum8):
+    """Tuya bulb type."""
+
+    LED = 0x00
+    INCANDESCENT = 0x01
+    HALOGEN = 0x02
+
+
+class F000LevelControlCluster(NoManufacturerCluster, LevelControl):
+    """LevelControlCluster that reports to attrid 0xF000."""
+
+    server_commands = LevelControl.server_commands.copy()
+    server_commands[TUYA_CUSTOM_LEVEL_COMMAND] = foundation.ZCLCommandDef(
+        "moveToLevelTuya",
+        (TuyaLevelPayload,),
+        is_manufacturer_specific=False,
+    )
+
+    attributes = LevelControl.attributes.copy()
+    attributes.update(
+        {
+            # 0xF000
+            TUYA_LEVEL_ATTRIBUTE: ("manufacturer_current_level", t.uint16_t),
+            # 0xFC02
+            TUYA_BULB_TYPE_ATTRIBUTE: ("bulb_type", TuyaBulbType),
+            # 0xFC03
+            TUYA_MIN_LEVEL_ATTRIBUTE: ("manufacturer_min_level", t.uint16_t),
+            # 0xFC04
+            TUYA_MAX_LEVEL_ATTRIBUTE: ("manufacturer_max_level", t.uint16_t),
+        }
+    )
+
+    # 0xF000 reported values are 10-1000, convert to 0-254
+    def _update_attribute(self, attrid, value):
+        if attrid == TUYA_LEVEL_ATTRIBUTE:
+            self.debug(
+                "Getting brightness %s",
+                value,
+            )
+            value = (value + 4 - 10) * 254 // (1000 - 10)
+            attrid = 0x0000
+
+        super()._update_attribute(attrid, value)
+
+    async def command(
+        self,
+        command_id: Union[foundation.Command, int, t.uint8_t],
+        *args,
+        manufacturer: Optional[Union[int, t.uint16_t]] = None,
+        expect_reply: bool = True,
+        tsn: Optional[Union[int, t.uint8_t]] = None,
+    ):
+        """Override the default Cluster command."""
+        self.debug(
+            "Sending Cluster Command. Cluster Command is %x, Arguments are %s",
+            command_id,
+            args,
+        )
+        # move_to_level, move, move_to_level_with_on_off
+        if command_id in (0x0000, 0x0001, 0x0004):
+            # convert dim values to 10-1000
+            brightness = args[0] * (1000 - 10) // 254 + 10
+            self.debug(
+                "Setting brightness to %s",
+                brightness,
+            )
+            return await super().command(
+                TUYA_CUSTOM_LEVEL_COMMAND,
+                TuyaLevelPayload(level=brightness, transtime=0),
+                manufacturer=manufacturer,
+                expect_reply=expect_reply,
+                tsn=tsn,
+            )
```

Comment:
Hmm that doesn't work because then this Class would also show up as cluster in the dimmers of other devices. `F000LevelControlCluster` has to inherit in some way from `CustomCluster` and `NoManufacturerCluster` does that for us (including setting `manufacturer=foundation.ZCLHeader.NO_MANUFACTURER_ID,`).

Reply:
> this Class would also show up as cluster in the dimmers of other devices

That sentence not make sense to me. A quirk shouldn't modify other devices that don't use the quirk.
Trying to figure out how can be done I believe that some code like this could make the described behaviour:
```python

    # attributes = LevelControl.attributes.copy()
    attributes.update(
        {
            # 0xF000
            TUYA_LEVEL_ATTRIBUTE: ("manufacturer_current_level", t.uint16_t),
        }
    )
```
If not making a copy of parent class attributes, maybe the code is modifying the parent attributes and changes for any device that uses the parent class 🤔 

In any case, your proposed code is fine to me.

Reply:
Hmm yeah I don’t know, but it did show up as level control cluster on this new dimmer I bought…